controllers/akodeploymentconfig/user/user_controller.go

```diff
@@ -310,10 +315,34 @@ func (r *AkoUserReconciler) getOrCreateAkoUserRole(roleTenantRef *string) (*mode
 		}
 		return r.aviClient.RoleCreate(role)
 	}
-	// else return role or error
+	if err == nil {
+		return r.ensureAkoUserRole()
+	}
 	return role, err
 }
 
+// ensureAkoUserRole ensure ako-essential-role has the latest permission
+func (r *AkoUserReconciler) ensureAkoUserRole() (*models.Role, error) {
+	role, err := r.aviClient.RoleGetByName(akoov1alpha1.AkoUserRoleName)
+	if err != nil {
+		return role, err
+	}
+	// check if role needs to be sync
+	needSync := false
+	for i, permission := range role.Privileges {
```

Comment:
I think we should check for different permission instead of hardcoding each permission type ere.
Then if the permission we want is lower than the expected, error out.
Otherwise we have to change two places everytime